ipn/ipnlocal/profiles.go

```diff
@@ -51,22 +51,39 @@ func (pm *profileManager) SetCurrentUser(uid string) error {
 	if pm.currentUserID == uid {
 		return nil
 	}
+	prev := pm.currentUserID
 	pm.currentUserID = uid
-	cpk := ipn.CurrentProfileKey(uid)
-	if b, err := pm.store.ReadState(cpk); err == nil {
-		pk := ipn.StateKey(string(b))
-		prefs, err := pm.loadSavedPrefs(pk)
-		if err != nil {
-			return err
-		}
-		pm.currentProfile = pm.findProfileByKey(pk)
-		pm.prefs = prefs
-		pm.isNewProfile = false
-	} else if err == ipn.ErrStateNotExist {
+	if uid == "" && prev != "" {
+		// This is a local user logout, or app shutdown.
+		// Clear the current profile.
+		pm.NewProfile()
+		return nil
+	}
+	pm.currentUserID = uid
+
+	// Read the CurrentProfileKey from the store which stores
+	// the selected profile for the current user.
+	b, err := pm.store.ReadState(ipn.CurrentProfileKey(uid))
+	if err == ipn.ErrStateNotExist || len(b) == 0 {
```

Comment:
not in the current code, but I don't want to crash in case this ever changes. It felt like a safe thing to do as not having any value there means we can't actually continue down the function and should return early.